lib/gym/commands_executor.rb

```diff
@@ -0,0 +1,56 @@
+module Gym
+  module CommandsExecutor
+    #####################################################
+    # @!group Actually executing the commands
+    #####################################################
+
+    # @param command [String] The command to be executed
+    # @param print_all [Boolean] Do we want to print out the command output while building?
+    #   If set to false, nothing will be printed
+    # @param error [Block] A block that's called if an error occurs
+    # @return [String] All the output as string
+    def self.execute(command: nil, print_all: false, error: nil)
```

Comment:
Looks like every invocation of execute is preceeded by a call to runner#print_command.
Maybe we could move that method in there, adding a description parameter that if not nil will result in triggering the print as well ?